src/mesh/voxel_cloud.rs

```diff
@@ -214,6 +213,191 @@ impl ScalarField {
+            return ScalarField::from_scalar_field(
+                source_scalar_field,
+                volume_value_range,
+                voxel_dimensions,
+            );
+        }
+
+        // Make a bounding box of the source scalar field's volume. This will be
+        // the volume to be scanned for voxels.
+        if let Some(source_sf_bounding_box) =
+            source_scalar_field.bounding_box_volume_cartesian_space(volume_value_range)
+        {
+            let vector_to_origin = Vector3::zeros() - source_sf_bounding_box.center().coords;
+
+            // Transform the source mesh volume bounding box and calculate a new
+            // bounding box that will encompass the transformed source scalar
+            // field.
+            let transformation_translate_to_origin = Matrix4::new_translation(&vector_to_origin);
+            let transformation_rotate_scale =
+                Matrix4::from(*rotate) * Matrix4::new_nonuniform_scaling(scale);
+
+            let source_sf_bounding_box_corners = source_sf_bounding_box.corners();
+            let transformed_source_sf_bounding_box_corners =
+                source_sf_bounding_box_corners.iter().map(|v| {
+                    let v1 = transformation_translate_to_origin.transform_point(v);
+                    transformation_rotate_scale.transform_point(&v1)
+                });
+
+            let transformed_bounding_box =
+                BoundingBox::from_points(transformed_source_sf_bounding_box_corners)
+                    .expect("No source bounding box");
+
+            // New scalar field that can encompass the transformed source scalar
+            // field's mesh.
+            let mut target_scalar_field = ScalarField::from_bounding_box_cartesian_space(
+                &transformed_bounding_box,
+                &voxel_dimensions,
+            );
+
+            // Transform the target voxels inverse to the user transformation so
+            // that it is possible to sample the source scalar field.
+            if let Ok(reversed_user_transformation) =
+                transformation_rotate_scale.pseudo_inverse(f32::EPSILON)
+            {
+                for (one_dimensional_target, voxel_target) in
+                    target_scalar_field.voxels.iter_mut().enumerate()
+                {
+                    let cartesian_coordinate_target = one_dimensional_to_cartesian_coordinate(
+                        one_dimensional_target,
+                        &target_scalar_field.block_start,
+                        &target_scalar_field.block_dimensions,
+                        &target_scalar_field.voxel_dimensions,
+                    );
+
+                    // Transform each new voxel inverse to the user
+                    // transformation.
+                    let transformed_voxel_center_cartesian = reversed_user_transformation
+                        .transform_point(&(cartesian_coordinate_target - vector_to_origin));
```

Comment:
I suspect this line to cause the offset

Reply:
Not sure I follow

Reply:
Btw not sure if you intended to fix it, but the shift still happens.

Reply:
shift happens

Reply:
can you please share your testing parameters and a screenshot of a faulty result?
i believed i fixed the issue, so i think i should test more thoroughly

Reply:
Okay, will do so (when I get to this PR in May).